katharsis-core/src/test/java/io/katharsis/resource/mock/models/Task.java

```diff
@@ -77,6 +73,15 @@ public void setName(@SuppressWarnings("SameParameterValue") String name) {
         this.name = name;
     }
 
+    public String getCategory() {
+        return category;
+    }
+
+    public void setCategory(String category) {
+    	if(category == null) throw new IllegalArgumentException("Category cannot be set to null!");
```

Comment:
I agree with you in general. At first I created a new Resource + Repository for this test, however I wanted to ensure that no other existing tests broke. The only way to do that without writing/copying new tests was to modify an already heavily tested Resource. Hence the use of `Task`.